clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -79,12 +80,20 @@
    * @throws IOException if an I/O error occurs
    */
   public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
+    this(fileId, options, FileSystemContext.INSTANCE);
+  }
+
+  // Test-only constructor for mocking the file system context. Whitebox cannot be used here because
+  // the context is used before the constructor finishes.
+  FileOutStream(long fileId, OutStreamOptions options, FileSystemContext context)
+      throws IOException {
     mFileId = fileId;
     mNonce = ClientContext.getRandomNonNegativeLong();
     mBlockSize = options.getBlockSizeBytes();
     mTachyonStorageType = options.getTachyonStorageType();
     mUnderStorageType = options.getUnderStorageType();
-    mContext = FileSystemContext.INSTANCE;
+    mContext = context;
+    mBlockStoreContext = BlockStoreContext.INSTANCE;
```

Comment:
add mBlockStoreContext as a instance variable so that it can be mocked via WhiteBox